src/NuGet.Core/NuGet.ProjectManagement/Projects/MSBuildNuGetProject.cs

```diff
@@ -577,19 +577,17 @@ public IReadOnlyList<PackageSpec> GetPackageSpecsForRestore(ExternalProjectRefer
             {
                 foreach (var reference in references)
                 {
-                    MSBuildRestoreUtility.AddMSBuildProjectReference(
-                        packageSpec,
-                        new ProjectRestoreReference
-                        {
-                            ProjectUniqueName = reference.UniqueName,
-                            ProjectPath = reference.MSBuildProjectPath
-                        },
-                        new LibraryDependency
-                        {
-                            LibraryRange = new LibraryRange(
-                                reference.UniqueName,
-                                LibraryDependencyTarget.ExternalProject)
-                        });
+                    // This reference applies to all frameworks
+                    // Include/exclude flags are not possible for this project type
+                    var projectReference = new ProjectRestoreReference
+                    {
+                        ProjectUniqueName = reference.UniqueName,
+                        ProjectPath = reference.MSBuildProjectPath
+                    };
+
+                    var group = new ProjectRestoreMetadataFrameworkInfo(MSBuildNuGetProjectSystem.TargetFramework);
```

Comment:
>ProjectRestoreMetadataFrameworkInfo [](start = 36, length = 35)

Same as BuildIntegrated, shouldn't this only be initialized once  per framework (once total in this case since packages.config)